code/modules/projectiles/gun.dm

```diff
@@ -1608,6 +1610,9 @@ not all weapons use normal magazines etc. load_into_chamber() itself is designed
 		user = gun_user
 
 	if(flags_gun_features & GUN_AMMO_COUNTER && current_mag)
+		// toggleable spam control.
+		if(user.client.prefs.toggle_prefs & TOGGLE_AMMO_DISPLAY_TYPE && current_mag.current_rounds % 5 != 0 && current_mag.current_rounds > 15 && gun_firemode != GUN_FIREMODE_BURSTFIRE)
```

Comment:
```suggestion
		if(user.client.prefs.toggle_prefs & TOGGLE_AMMO_DISPLAY_TYPE && gun_firemode == GUN_FIREMODE_SEMIAUTO && current_mag.current_rounds % 5 != 0 && current_mag.current_rounds > 15)
```
Equal to because by not working for full auto I mean that you would have to change the logic to be when you *pass* the threshold to work properly. At that point it could be used for all fire modes, but done this way it can only work for semi auto.

Reply:
Sorry, I was going on the previous suggestion
> You should have this either send a message ever we pass the threshold, or just always on a burst. You could also convey that the preference only works when the mag count is above 15.